crates/ink/ir/src/ir/event/config.rs

```diff
@@ -44,27 +48,56 @@ impl TryFrom<ast::AttributeArgs> for EventConfig {
                         "expected a bool literal for `anonymous` ink! event item configuration argument",
                     ));
                 }
+            } else if arg.name.is_ident("signature_topic") {
+                if anonymous.is_some() {
+                    return Err(format_err_spanned!(
+                        arg,
+                        "cannot specify `signature_topic` with `anonymous` in ink! event item configuration argument",
+                    ));
+                }
+
+                if let Some(lit_str) = signature_topic {
+                    return Err(duplicate_config_err(lit_str, arg, "anonymous", "event"));
+                }
+                if let ast::MetaValue::Lit(syn::Lit::Str(lis_str)) = &arg.value {
+                    signature_topic = Some(lis_str.clone())
+                } else {
+                    return Err(format_err_spanned!(
+                        arg,
+                        "expected a bool literal for `anonymous` ink! event item configuration argument",
+                    ));
+                }
             } else {
                 return Err(format_err_spanned!(
                     arg,
                     "encountered unknown or unsupported ink! storage item configuration argument",
                 ));
             }
         }
+
         Ok(EventConfig::new(
             anonymous.map(|lit_bool| lit_bool.value).unwrap_or(false),
+            signature_topic.map(|lit_str| lit_str.value()),
         ))
     }
 }
 
 impl EventConfig {
     /// Construct a new [`EventConfig`].
-    pub fn new(anonymous: bool) -> Self {
-        Self { anonymous }
+    pub fn new(anonymous: bool, signature_topic_hash: Option<String>) -> Self {
+        Self {
+            anonymous,
+            signature_topic_hash,
+        }
     }
 
     /// Returns the anonymous configuration argument.
     pub fn anonymous(&self) -> bool {
         self.anonymous
     }
+
+    /// Returns the manually specified signature topic.
+    pub fn signature_topic_hash(&self) -> Option<String> {
+        self.signature_topic_hash.clone()
```

Comment:
Nitpick: not a big deal at all but couldn't this return `Option<&str>` and then just do `.as_ref()`.